README.md
# Sphere Bottom Navigation Bar

**Sphere Bottom Navigation Bar is flutter package, You can create beautiful awesome degined bottom navigation bar with importing this library**

## Getting Started

### 1.  Add the dependencies to [pubspec.yaml]
```yaml
dependencies:
   sphere_bottom_navigation_bar: ^1.0.0
```
---
###  2.  Install it
You can install packages from the command line:
```shell
$ flutter packages get
```
---


### 3. Import package
```dart
import  'package:sphere_bottom_navigation_bar/sphere_bottom_navigation_bar.dart';
```
---
###  4.  Example: 

```dart
bottomNavigationBar: SphereBottomNavigationBar(  
  defaultSelectedItem: 0,  
  sheetRadius: BorderRadius.only(  
      topLeft: Radius.circular(20), topRight: Radius.circular(20)),  
  onItemPressed: (index) => setState(() {  
        print('Selected item is - $index');  
      }),   
  navigationItems: [  
    BuildNavigationItem(  
      tooltip: 'Home',  
  itemColor: Color(0xFFFFD6B2),  
  icon: Icon(Icons.home),  
  selectedItemColor: Color(0xFFFFD6B2),  
  ),  
  BuildNavigationItem(  
      tooltip: 'Chat',  
  itemColor: Color(0xFFB2F4FF),  
  icon: Icon(Icons.chat),  
  selectedItemColor: Color(0xFFB2F4FF),  
  ),  
  BuildNavigationItem(  
      tooltip: 'Peoples',  
  itemColor: Color(0xFFCDB2FF),  
  icon: Icon(Icons.people),  
  selectedItemColor: Color(0xFFCDB2FF),  
  ),  
  BuildNavigationItem(  
        tooltip: 'Settings',  
  itemColor: Color(0xFFFFB2D9),  
  icon: Icon(Icons.settings),  
  selectedItemColor: Color(0xFFFFB2D9)),  
  ],  
),
```

# Screenshots 


<div> <img src="https://github.com/PremRajpurohit/sphere_bottom_navigation_bar/blob/master/screenshots/screenshot1.png" alt="drawing" width="200"/> <img src="https://github.com/PremRajpurohit/sphere_bottom_navigation_bar/blob/master/screenshots/screenshot2.png" alt="drawing" width="200"/> <img src="https://github.com/PremRajpurohit/sphere_bottom_navigation_bar/blob/master/screenshots/screenshot3.png" alt="drawing" width="200"/> </div>




CHANGELOG.md
## [0.0.1] - SAT 23 FEB 2019

* Publish

## [0.0.2] - SAT 23 FEB 2019

*Fixed some errors

## [0.0.3] - SAT 23 FEB 2019

*Fixed Some Errors and README.MD

## [1.0.0] - SAT 16 MARCH 2019

*Fixed Some Errors and And ADDED MIT License

lib/sphere_bottom_navigation_bar.dart
library sphere_bottom_navigation_bar;

import 'package:flutter/material.dart';

class BuildNavigationItem {
  final Icon icon;
  final Color selectedItemColor;
  final Color itemColor;
  final String tooltip;
  BuildNavigationItem(
      {@required this.icon,
      @required this.tooltip,
      this.selectedItemColor = Colors.blue,
      this.itemColor}) {
    assert(icon != null);
  }
}

class SphereBottomNavigationBar extends StatefulWidget {
  final int defaultSelectedItem;
  final Color sheetBackgroundColor;
  final BorderRadius sheetRadius;
  final List<BuildNavigationItem> navigationItems;
  final ValueChanged<int> onItemPressed;
  final ValueChanged<int> onItemLongPressed;

  SphereBottomNavigationBar(
      {Key key,
      this.defaultSelectedItem,
      this.sheetBackgroundColor,
      this.sheetRadius,
      this.onItemLongPressed,
      @required this.navigationItems,
      @required this.onItemPressed}) {
    assert(onItemPressed != null, 'You must implement onItemPressed ');
  }

  @override
  _SphereBottomNavigationBarState createState() {
    return _SphereBottomNavigationBarState(
        items: navigationItems,
        sheetRadius: sheetRadius,
        onItemLongPressed: onItemLongPressed,
        sheetBackgroundColor: sheetBackgroundColor,
        defaultSelectedItem: defaultSelectedItem,
        onItemPressed: onItemPressed);
  }
}

class _SphereBottomNavigationBarState extends State<SphereBottomNavigationBar> {
  final int defaultSelectedItem;
  List<BuildNavigationItem> items;
  int selectedItemIndex;
  BorderRadius sheetRadius;
  var sheetHieght;
  var sheetWidth;
  Color sheetBackgroundColor;
  ValueChanged<int> onItemLongPressed;
  ValueChanged<int> onItemPressed;

  _SphereBottomNavigationBarState(
      {@required this.items,
      this.onItemLongPressed,
      this.defaultSelectedItem = 0,
      this.sheetRadius,
      this.sheetBackgroundColor,
      //this.iconSize,
      @required this.onItemPressed}) {
    selectedItemIndex = defaultSelectedItem;
    assert(items.length > 1, 'Atleast 2 item required. ');
    assert(items.length <= 5, 'You can add Maximum 5 Item');
  }

  Widget itemBuilder(BuildNavigationItem item, bool isSelected) {
    var containerHieght = ((sheetHieght + sheetWidth) / 8);
    var containerWidth = ((sheetHieght + sheetWidth) / 8);
    //print('Container hieght $containerHieght');
    //print('Container Width $containerWidth');
    var tooltip =
        item.tooltip == null ? selectedItemIndex.toString() : item.tooltip;
    return Padding(
      padding: const EdgeInsets.only(left: 6, right: 6),
      child: Tooltip(
        verticalOffset: 50,
        message: '$tooltip',
        child: Container(
          width: containerWidth,
          height: containerHieght,
          decoration: BoxDecoration(
            color: isSelected
                ? item.selectedItemColor.withOpacity(1)
                : item.itemColor.withOpacity(0.4),
            borderRadius: BorderRadius.all(Radius.circular(150)),
          ),
          child: Center(
            child: ListView(
              shrinkWrap: true,
              physics: NeverScrollableScrollPhysics(),
              scrollDirection: Axis.horizontal,
              children: <Widget>[
                Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  crossAxisAlignment: CrossAxisAlignment.center,
                  children: <Widget>[
                    IconTheme(
                      data: IconThemeData(
                          size: ((containerHieght + containerWidth) / 5),
                          color: isSelected
                              ? Colors.white
                              : item.itemColor == null
                                  ? Colors.white
                                  : item.itemColor),
                      child: item.icon,
                    ),
                  ],
                )
              ],
            ),
          ),
        ),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    sheetHieght = MediaQuery.of(context).size.height / 10;
    sheetWidth = MediaQuery.of(context).size.width;
    sheetBackgroundColor = (sheetBackgroundColor == null)
        ? Theme.of(context).bottomAppBarColor
        : sheetBackgroundColor;
    return Container(
      width: sheetWidth,
      height: sheetHieght,
      color: Colors.transparent,
      child: Container(
        padding: EdgeInsets.only(top: 6),
        width: sheetWidth,
        height: sheetHieght,
        decoration: BoxDecoration(
            color: sheetBackgroundColor,
            borderRadius: sheetRadius,
            boxShadow: [BoxShadow(color: Colors.black12, blurRadius: 3)]),
        child: Row(
          mainAxisAlignment: MainAxisAlignment.spaceAround,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: items.map((item) {
            var index = items.indexOf(item);
            return GestureDetector(
              onTap: () {
                onItemPressed(index);
                setState(() {
                  selectedItemIndex = index;
                });
              },
              onLongPress: () {
                onItemLongPressed(index);
                setState(() {
                  selectedItemIndex = index;
                });
              },
              child: itemBuilder(item, selectedItemIndex == index),
            );
          }).toList(),
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:sphere_bottom_navigation_bar/sphere_bottom_navigation_bar.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      debugShowMaterialGrid: false,
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({
    Key key,
  }) : super(key: key);
  @override
  MyHomePageState createState() => MyHomePageState();
}

class MyHomePageState extends State<MyHomePage> {
  Color backgroudColor = Colors.white;
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: backgroudColor,
      appBar: AppBar(
        backgroundColor: backgroudColor,
        title: Center(child: Text('Sphere Bottom Navigation')),
        elevation: 0,
      ),
      bottomNavigationBar: SphereBottomNavigationBar(
        defaultSelectedItem: 0,
        sheetRadius: BorderRadius.only(
            topLeft: Radius.circular(20), topRight: Radius.circular(20)),
        onItemPressed: (index) => setState(() {
              print('Selected item is - $index');
              switch (index) {
                case 0:
                  backgroudColor = Color(0xFFFFD6B2);
                  break;
                case 1:
                  backgroudColor = Color(0xFFB2F4FF);
                  break;
                case 2:
                  backgroudColor = Color(0xFFCDB2FF);
                  break;
                case 3:
                  backgroudColor = Color(0xFFFFB2D9);
                  break;
                default:
              }
            }),
        onItemLongPressed: (index) => setState(() {
              backgroudColor = Color(0xFF44D6B2);
            }),
        navigationItems: [
          BuildNavigationItem(
            tooltip: 'Home',
            itemColor: Color(0xFFFFD6B2),
            icon: Icon(Icons.home),
            selectedItemColor: Color(0xFFFFD6B2),
          ),
          BuildNavigationItem(
            tooltip: 'Chat',
            itemColor: Color(0xFFB2F4FF),
            icon: Icon(Icons.chat),
            selectedItemColor: Color(0xFFB2F4FF),
          ),
          BuildNavigationItem(
            tooltip: 'Peoples',
            itemColor: Color(0xFFCDB2FF),
            icon: Icon(Icons.people),
            selectedItemColor: Color(0xFFCDB2FF),
          ),
          BuildNavigationItem(
              tooltip: 'Settings',
              itemColor: Color(0xFFFFB2D9),
              icon: Icon(Icons.settings),
              selectedItemColor: Color(0xFFFFB2D9)),
        ],
      ),
      body: Container(
        color: backgroudColor,
      ),
    );
  }
}
